File: main_module/src/main/java/com/mobile/main_module/login/ui/LoginActivity.java
package com.mobile.main_module.login.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.mobile.common_library.base.BaseMvpAppActivity;
import com.mobile.common_library.base.IBaseView;
import com.mobile.common_library.manager.ActivityPageManager;
import com.mobile.common_library.manager.DeviceManager;
import com.mobile.common_library.manager.LogManager;
import com.mobile.main_module.R;
import com.mobile.main_module.login.bean.GetVerificationCode;
import com.mobile.main_module.login.bean.LoginResponse;
import com.mobile.main_module.login.presenter.LoginPresenterImpl;
import com.mobile.main_module.login.view.ILoginView;
import com.mobile.main_module.main.MainActivity;

import androidx.appcompat.widget.Toolbar;

@Route(path = "/main_module/login")
public class LoginActivity extends BaseMvpAppActivity<IBaseView, LoginPresenterImpl>
	implements ILoginView {
	
	private static final String      TAG = "LoginActivity";
	private              Toolbar     toolbar;
	private              FrameLayout layoutBack;
	private              ImageView   imvBack;
	private              EditText    edtAccountNumber;
	private              EditText    edtPassword;
	private              FrameLayout layoutVerificationCode;
	private              EditText    edtVerificationCode;
	private              TextView    tevGetAuthCode;
	private              TextView    tevLogin;
	private              TextView    tevJumpToRegister;
	
	private String accountNumber;
	private String verificationCode;
	private String phoneDevice;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
	}
	
	@Override
	protected int initLayoutId() {
		return R.layout.activity_login;
	}
	
	@Override
	protected void initData() {
		ActivityPageManager.getInstance().finishAllActivityExcept(LoginActivity.class);
	}
	
	@Override
	protected void initViews() {
		toolbar = (Toolbar) findViewById(R.id.toolbar);
		layoutBack = (FrameLayout) findViewById(R.id.layout_back);
		imvBack = (ImageView) findViewById(R.id.imv_back);
		edtAccountNumber = (EditText) findViewById(R.id.edt_account_number);
		edtPassword = (EditText) findViewById(R.id.edt_password);
		layoutVerificationCode = (FrameLayout) findViewById(R.id.layout_verification_code);
		edtVerificationCode = (EditText) findViewById(R.id.edt_verification_code);
		tevGetAuthCode = (TextView) findViewById(R.id.tev_get_auth_code);
		tevLogin = (TextView) findViewById(R.id.tev_login);
		tevJumpToRegister = (TextView) findViewById(R.id.tev_jump_to_register);
		
		addContentView(loadView, layoutParams);
		setToolbar(true, R.color.color_FFFFFFFF);
		
		layoutBack.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				finish();
			}
		});
		tevGetAuthCode.setOnClickListener(view -> {
			getAuthCode();
		});
		tevLogin.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				initLoginWithAuthCode();
			}
		});
		tevJumpToRegister.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				startActivity(RegisterActivity.class);
			}
		});
	}
	
	@Override
	protected void initLoadData() {
		edtAccountNumber.setText("13513313214");
	}
	
	@Override
	protected LoginPresenterImpl attachPresenter() {
		return new LoginPresenterImpl(this);
	}
	
	@Override
	public void showLoading() {
		if (loadView != null && !loadView.isShown()) {
			loadView.setVisibility(View.VISIBLE);
			loadView.start();
		}
	}
	
	@Override
	public void hideLoading() {
		if (loadView != null && loadView.isShown()) {
			loadView.stop();
			loadView.setVisibility(View.GONE);
		}
	}
	
	@Override
	protected void onNewIntent(Intent intent) {
		super.onNewIntent(intent);
		LogManager.i(TAG, "onNewIntent");
	}
	
	@Override
	public void getAuthCodeSuccess(GetVerificationCode.DataDTO success) {
		showToast(success.getContent(), true);
	}
	
	@Override
	public void getAuthCodeError(String error) {
		showToast(error, true);
	}
	
	@Override
	public void loginWithAuthCodeSuccess(LoginResponse.DataDTO success) {
		//		showToast(success.getUserName(), true);
		startActivity(MainActivity.class);
		finish();
	}
	
	@Override
	public void loginWithAuthCodeError(String error) {
		showToast(error, true);
	}
	
	private void getAuthCode() {
		accountNumber = edtAccountNumber.getText().toString();
		//		verificationCode = edtVerificationCode.getText().toString();
		//		phoneDevice = UIUtils.getDeviceUUid().toString();
		
		bodyParams.clear();
		bodyParams.put("account", accountNumber);
		presenter.getAuthCode(this, bodyParams);
	}
	
	private void initLoginWithAuthCode() {
		accountNumber = edtAccountNumber.getText().toString();
		verificationCode = edtVerificationCode.getText().toString();
		phoneDevice = DeviceManager.getDeviceUUid();
		
		bodyParams.clear();
		bodyParams.put("account", accountNumber);
		bodyParams.put("captcha", verificationCode);
		bodyParams.put("type", "1");//1 APP
		bodyParams.put("phoneDevice", phoneDevice);
		presenter.loginWithAuthCode(this, bodyParams);
	}
	
}

File: module_main/src/main/java/com/phone/module_main/login/ui/LoginActivity.kt
package com.phone.module_main.login.ui

import android.content.Intent
import android.text.TextUtils
import android.view.View
import android.widget.EditText
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.widget.Toolbar
import com.alibaba.android.arouter.facade.annotation.Route
import com.alibaba.android.arouter.launcher.ARouter
import com.phone.library_common.base.BaseMvpRxAppActivity
import com.phone.library_common.base.IBaseView
import com.phone.library_common.common.ConstantData
import com.phone.library_common.manager.*
import com.phone.library_common.manager.SystemManager.getSystemId
import com.phone.module_main.R
import com.phone.module_main.login.DataGetVerification
import com.phone.module_main.login.DataLogin
import com.phone.module_main.login.presenter.LoginPresenterImpl
import com.phone.module_main.login.view.ILoginView

@Route(path = ConstantData.Route.ROUTE_LOGIN)
class LoginActivity : BaseMvpRxAppActivity<IBaseView, LoginPresenterImpl>(), ILoginView {

    private val TAG = LoginActivity::class.java.simpleName
    private var toolbar: Toolbar? = null
    private var layoutBack: FrameLayout? = null
    private var imvBack: ImageView? = null
    private var imvHeadPortrait: ImageView? = null
    private var edtUserId: EditText? = null
    private var edtPassword: EditText? = null
    private var layoutVerificationCode: FrameLayout? = null
    private var edtVerificationCode: EditText? = null
    private var tevGetAuthCode: TextView? = null
    private var tevLogin: TextView? = null
    private var tevJumpToRegister: TextView? = null

    private var userId: String? = null
    private var password: String? = null
    private var verificationCode: String? = null
    private var phoneDevice: String? = null

    override fun initLayoutId(): Int {
        return R.layout.main_activity_login
    }

    override fun initData() {
        ActivityPageManager.instance().finishAllActivityExcept(LoginActivity::class.java)
    }

    override fun initViews() {
        toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        layoutBack = findViewById<View>(R.id.layout_back) as FrameLayout
        imvBack = findViewById<View>(R.id.imv_back) as ImageView
        imvHeadPortrait = findViewById<View>(R.id.imv_head_portrait) as ImageView
        edtUserId = findViewById<View>(R.id.edt_user_id) as EditText
        edtPassword = findViewById<View>(R.id.edt_password) as EditText
        layoutVerificationCode = findViewById<View>(R.id.layout_verification_code) as FrameLayout
        edtVerificationCode = findViewById<View>(R.id.edt_verification_code) as EditText
        tevGetAuthCode = findViewById<View>(R.id.tev_get_auth_code) as TextView
        tevLogin = findViewById<View>(R.id.tev_login) as TextView
        tevJumpToRegister = findViewById<View>(R.id.tev_jump_to_register) as TextView
        setToolbar(true, R.color.color_FFFFFFFF)
        imvBack?.setColorFilter(ResourcesManager.getColor(R.color.color_000000))
        layoutBack?.setOnClickListener { finish() }
        imvHeadPortrait?.let {
            ImageLoaderManager.displayRound(mRxAppCompatActivity, it, R.mipmap.ic_launcher_round)
        }
        tevGetAuthCode?.setOnClickListener { view: View? -> getAuthCode() }
        tevLogin?.setOnClickListener { //                initLoginWithAuthCode()
            initLogin()
        }
        tevJumpToRegister?.setOnClickListener {
            ARouter.getInstance().build(ConstantData.Route.ROUTE_REGISTER)
                .navigation()
        }
    }

    override fun initLoadData() {
        edtUserId?.setText("13510001000")
        edtPassword?.setText("12345678")
    }

    override fun attachPresenter(): LoginPresenterImpl {
        return LoginPresenterImpl(this)
    }

    override fun showLoading() {
        if (!mLoadView.isShown) {
            mLoadView.visibility = View.VISIBLE
            mLoadView.start()
        }
    }

    override fun hideLoading() {
        if (mLoadView.isShown) {
            mLoadView.stop()
            mLoadView.visibility = View.GONE
        }
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        LogManager.i(TAG, "onNewIntent")
    }

    override fun getAuthCodeSuccess(success: DataGetVerification) {
        if (!mRxAppCompatActivity.isFinishing) {
            showToast(success.content, true)
            hideLoading()
        }
    }

    override fun getAuthCodeError(error: String) {
        if (!mRxAppCompatActivity.isFinishing) {
            showToast(error, true)
            hideLoading()
        }
    }

    override fun loginWithAuthCodeSuccess(success: DataLogin) {
        if (!mRxAppCompatActivity.isFinishing) {
            hideLoading()
            customStartActivity()
        }
    }

    override fun loginWithAuthCodeError(error: String) {
        if (!mRxAppCompatActivity.isFinishing) {
            showToast(error, true)
            hideLoading()
        }
    }

    override fun loginSuccess(success: String) {
        if (!mRxAppCompatActivity.isFinishing) {
            hideLoading()
            customStartActivity()
        }
    }

    override fun loginError(error: String) {
        if (!mRxAppCompatActivity.isFinishing) {
            showToast(error, true)
            hideLoading()
        }
    }

    private fun customStartActivity() {
        SharedPreferencesManager.put("isLogin", true)
        ARouter.getInstance().build(ConstantData.Route.ROUTE_MAIN)
            .navigation()
        finish()
    }

    private fun getAuthCode() {
        userId = edtUserId?.text.toString()
        //		verificationCode = edtVerificationCode.getText().toString()
        //		phoneDevice = UIUtils.getDeviceUUid().toString()
        userId?.let {
            mBodyParams.clear()
            mBodyParams["account"] = it
            presenter.getAuthCode(this, mBodyParams)
        }
    }

    private fun initLoginWithAuthCode() {
        userId = edtUserId?.text.toString()
        verificationCode = edtVerificationCode?.text.toString()
        phoneDevice = getSystemId()
        if (TextUtils.isEmpty(userId)) {
            showToast(ResourcesManager.getString(R.string.please_enter_one_user_name), true)
            return
        }
        if (TextUtils.isEmpty(verificationCode)) {
            showToast(ResourcesManager.getString(R.string.please_input_a_password), true)
            return
        }
        if (TextUtils.isEmpty(phoneDevice)) {
            showToast(ResourcesManager.getString(R.string.please_input_a_password), true)
            return
        }
        showLoading()
        if (RetrofitManager.isNetworkAvailable()) {
            mBodyParams.clear()
            mBodyParams["account"] = userId!!
            mBodyParams["captcha"] = verificationCode!!
            mBodyParams["type"] = "1" //1 APP
            mBodyParams["phoneDevice"] = phoneDevice!!
            presenter.loginWithAuthCode(this, mBodyParams)
        } else {
            showToast(
                ResourcesManager.getString(R.string.please_check_the_network_connection),
                true
            )
            hideLoading()
        }
    }

    private fun initLogin() {
        showLoading()
        if (RetrofitManager.isNetworkAvailable()) {
            userId = edtUserId?.text.toString()
            password = edtPassword?.text.toString()
            if (TextUtils.isEmpty(userId)) {
                showToast(ResourcesManager.getString(R.string.please_enter_one_user_name), true)
                return
            }
            if (TextUtils.isEmpty(password)) {
                showToast(ResourcesManager.getString(R.string.please_input_a_password), true)
                return
            }
            mBodyParams.clear()
            mBodyParams["userId"] = userId!!
            mBodyParams["password"] = password!!
            presenter.login(this, mBodyParams)
        } else {
            showToast(
                ResourcesManager.getString(R.string.please_check_the_network_connection),
                true
            )
            hideLoading()
        }
    }


}